cache/cache.go

```diff
@@ -0,0 +1,190 @@
+package cache
+
+import (
+	"container/list"
+	"sync"
+)
+
+// Cache is a bounded-size in-memory cache of sized items with a configurable eviction policy
+type Cache interface {
+	// Get retrieves items from the cache by key.
+	// If an item for a particular key is not found, its position in the result will be nil.
+	Get(keys ...string) []Item
```

Comment:
I don't usually love APIs like this, where the order of things in the returned list has an important correspondence to the input. Is there a big performance or usability benefit to accepting a list of strings?